File: examples/get_namespace.rs
//
// Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//

use objectscale_client::client::ManagementClient;

fn main() {
    let endpoint = "https://10.225.108.217:4443";
    let username = "root";
    let password = "Password123!";
    let insecure = true;

    let id = "luis_namespace";

    let mut client =
        ManagementClient::new(endpoint, username, password, insecure).expect("management client");
    let namespace = client.get_namespace(id).expect("get namespace");
    println!("Get namespace: {:?}", namespace);
}

File: python/examples/get_namespace.py
import objectscale_client

def main():
    endpoint = "https://10.225.108.217:4443"
    username = "root"
    password = "Password123!"
    insecure = True

    name = "luis_namespace"

    try:
        client = objectscale_client.client.ManagementClient(endpoint, username, password, insecure)
        namespace = client.get_namespace(name)
        print("Get namespace:", namespace)
    except Exception as e:
        print("Failed to get namespace:", e)


if __name__ == '__main__':
    main()
